liquibase-core/src/main/java/liquibase/resource/AbstractResource.java

```diff
@@ -60,4 +60,18 @@ public boolean equals(Object obj) {
         }
         return this.getUri().equals(((Resource) obj).getUri());
     }
+
+    /**
+     * Convenience method for computing the relative path in {@link #resolve(String)} implementations
+     */
+    protected String resolvePath(String other) {
+        return getPath() + "/" + other;
+    }
+
+    /**
+     * Convenience method for computing the relative path in {@link #resolveSibling(String)} implementations.
+     */
+    protected String resolveSiblingPath(String other) {
+        return getPath().replaceFirst("/[^/]*$", "") + "/" + other;
```

Comment:
This looks brittle. Is that a problem?

Reply:
I thought maybe it was OK? And since its a protected method, it can be overriden for subclasses where it doesn't quite work.

But what types of paths did you think it would be  a problem for?

Reply:
What about a path that uses backslashes instead of forward slashes?

Reply:
Coming up with examples of how to break this is not easy. That's why I used the Commons IO code when I did the first round on this, because I was relatively confident that they have come up with good solutions to these situations.

Reply:
That path should be the standardized "path in the resourceAccessor" so it should already just use `/` and not have any odd `..` or whatever in them. So shouldn't need to much fancy logic?